### PR TITLE
Send the last ping when the user disable send usage data

### LINKED ISSUE #2487: Know the number of users turn off "Send usage data"

### PR DESCRIPTION
**Pull request by cnevinc**
closes #2487

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt [hunk 1]
```diff
@@ -1032,15 +1041,43 @@ object TelemetryWrapper {
...
+
                 firebaseEvent.event(context)
             }
         }
 
+        private fun sendEventNow(event: TelemetryEvent) {
+            val telemetry = TelemetryHolder.get()
+            var focusEventBuilder: TelemetryPingBuilder? = null
+            // we only have FocusEventPing now
+            for (telemetryPingBuilder in telemetry.builders) {
```
Comment (mTwTm) [inline app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt]: What about builders.get(TelemetryEventPingBuilder.TYPE)

File: app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt [hunk 2]
```diff
@@ -254,7 +263,7 @@ object TelemetryWrapper {
                             resources.getString(R.string.pref_key_locale))
                     .setSettingsProvider(CustomSettingsProvider())
                     .setCollectionEnabled(telemetryEnabled)
-                    .setUploadEnabled(telemetryEnabled)
```
[the remark this answers is not in the document]
Comment (cnevinc) [reply app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt]: no. we'll always enable upload. Only stop collecting

File: app/src/main/java/org/mozilla/focus/settings/SettingsFragment.java [hunk 3]
```diff
@@ -153,9 +153,10 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, Strin
                     .replace(R.id.container, new SettingsFragment())
                     .commit();
             return;
-        } else {
+            // we'll handle the pref_key_telemetry by TelemetrySwitchPreference
+        } else if (!key.equals(getString(R.string.pref_key_telemetry))) {
```
[the remark this answers is not in the document]
Comment (cnevinc) [reply app/src/main/java/org/mozilla/focus/settings/SettingsFragment.java]: yes. pings for changing pref_key_telemetry will only be collected TelemetrySwitchPreference's OnCheckedChangeListener. See: https://github.com/mozilla-tw/Rocket/blob/727706e16306ef2a0723dde6188fa2c84dfd74b0/app/src/main/java/org/mozilla/focus/widget/TelemetrySwitchPreference.java#L54

File: app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt [hunk 4]
```diff
@@ -219,13 +219,21 @@ object TelemetryWrapper {
...
 
+        val key = resources.getString(R.string.pref_key_telemetry)
         preferences.edit()
-                .putBoolean(resources.getString(R.string.pref_key_telemetry), enabled)
+                .putBoolean(key, enabled)
                 .apply()
 
-        TelemetryHolder.get()
-                .configuration
-                .setUploadEnabled(enabled).isCollectionEnabled = enabled
```
[the remark this answers is not in the document]
Comment (cnevinc) [reply app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt]: yes. We'll always enable upload.

[the remark this answers is not in the document]
Comment (mTwTm) [reply app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt]: Oops I thought it was a map.

File: app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (mxlius) [reply app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt]: Should we keep this line and explicitly pass true?

**Pull request merged**